### PR TITLE
Fix hue init action

### PR DESCRIPTION
**Pull request by mikaylakonst**

### CODE REVIEW

File: hue/hue.sh [hunk 1]
```diff
@@ -20,34 +20,36 @@ ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
 
 # Only run on the master node of the cluster
 if [[ "${ROLE}" == 'Master' ]]; then
+  # Install hue
   apt-get update
+  apt-get install -t jessie-backports hue-common -y
```
Comment (pmkc) [inline hue/hue.sh]: Why not just install hue (with -t jessie-backports here) and remove the next line?

File: hue/hue.sh [hunk 2]
```diff
@@ -20,34 +20,36 @@ ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
...
 EOF
   sed -i '/<\/configuration>/e cat core-site-patch.xml' \
      /etc/hadoop/conf/core-site.xml
-  
- 
+
+
   cat > hdfs-site-patch.xml <<EOF
-  <property>
+  <property>/ha
```
Comment (pmkc) [inline hue/hue.sh]: typo?

File: hue/hue.sh [hunk 3]
```diff
@@ -56,20 +58,64 @@ EOF
...
   cat >> /etc/hue/conf/hue.ini <<EOF
     # Defaults to $HADOOP_MR1_HOME or /usr/lib/hadoop-0.20-mapreduce
-    hadoop_mapred_home=/usr/lib/hadoop-mapreduce 
+    hadoop_mapred_home=/usr/lib/hadoop-mapreduce
 EOF
-       
-  # Replace localhost with hostname.
+
+  # Fix webhdfs_url
+  sed -i 's/## webhdfs_url\=http:\/\/localhost:50070/webhdfs_url\=http:\/\/'"$(hdfs getconf -confKey  dfs.namenode.http-address)"'/' /etc/hue/conf/hue.ini
```
Comment (pmkc) [inline hue/hue.sh]: It's painful, but could you enforce line length (at most 100 prefreably 80 chars) breaking strings with \ and using ""

File: hue/hue.sh [hunk 4]
```diff
@@ -56,20 +58,64 @@ EOF
...
 
   # Restart HDFS
-  /usr/lib/hadoop/libexec/init-hdfs.sh
+  service hadoop-hdfs-namenode restart
+
+  # Make hive warehouse directory
+  hdfs dfs -mkdir /user/hive/warehouse
+
+  # Restart yarn resourcemanager
+  service hadoop-yarn-resourcemanager restart
```
Comment (pmkc) [inline hue/hue.sh]: just use 'systemctl restart hadoop-hdfs-namenode hadoop-yarn-resourcemanager mysql hue at the end' once

Or maybe mysql first if hue syncdb needs it.

File: hue/hue.sh [hunk 5]
```diff
@@ -20,34 +20,36 @@ ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
 
 # Only run on the master node of the cluster
 if [[ "${ROLE}" == 'Master' ]]; then
```
Comment (pmkc) [inline hue/hue.sh]: It's not usual in our init actions but exiting 0 if we are not on the master saves a lot of indenting.

Comment (mikaylakonst) [reply hue/hue.sh]: done

File: hue/hue.sh (hunk 1 above)
Comment (mikaylakonst) [reply hue/hue.sh]: done

File: hue/hue.sh (hunk 2 above)
Comment (mikaylakonst) [reply hue/hue.sh]: yep - removed

File: hue/hue.sh (hunk 4 above)
Comment (mikaylakonst) [reply hue/hue.sh]: Done
Still restart hue and mysql beforehand because hue syncdb needs it

File: hue/hue.sh [hunk 6]
```diff
@@ -17,108 +17,97 @@ set -x -e
...
-  <property>
-    <name>hadoop.proxyuser.hue.hosts</name>
-    <value>*</value>
-  </property>
-  <property>
-    <name>hadoop.proxyuser.hue.groups</name>
-    <value>*</value>
-  </property>
+if [[ "${ROLE}" != 'Master' ]]; then
+  set +x +e
```
Comment (pmkc) [inline hue/hue.sh]: This isn't necessary with exit (or ever)

File: hue/hue.sh (hunk 3 above)
Comment (mikaylakonst) [reply hue/hue.sh]: Done

**Pull request merged**